src/bmson.rs

```diff
@@ -316,9 +323,58 @@ pub struct KeyEvent {
 
 /// Beatoraja implementation of invisible key channel.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct KeyChannel {
+pub struct KeyChannel<'a> {
     /// Name of the key sound file.
-    pub name: String,
+    pub name: Cow<'a, str>,
     /// Invisible key notes.
     pub notes: Vec<KeyEvent>,
 }
+
+/// Parse a BMSON file from JSON string.
+///
+/// This function provides a convenient way to parse a BMSON file in one step.
+/// It uses `serde_path_to_error` internally to provide detailed error information
+/// about the path to problematic fields in the JSON.
+///
+/// # Errors
+///
+/// Returns a `serde_path_to_error::Error<serde_json::Error>` if the JSON is malformed or contains invalid data.
+/// The error will include the path to the problematic field in the JSON.
+pub fn parse_bmson(json: &str) -> Result<Bmson<'_>, serde_path_to_error::Error<serde_json::Error>> {
+    // First try to parse with serde_path_to_error for better error reporting
+    let jd = &mut serde_json::Deserializer::from_str(json);
+    let result: Result<Bmson, _> = serde_path_to_error::deserialize(jd);
+
+    result
+}
+
+#[cfg(feature = "diagnostics")]
+impl ToAriadne for serde_path_to_error::Error<serde_json::Error> {
```

Comment:
I think we should not impl `ToAriadne` for `serde_path_to_error::Error`. Because it is a structural error information. Users will be confused with the meaningless span and line diagnosis.